### PR TITLE
Module Generative Anthropic

### PR DESCRIPTION
**Pull request by cdpierse**
### What's being changed:

This PR adds a generative module integration for the Anthropic Claude series of models. 

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: modules/generative-anthropic/config/class_settings.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+func (ic *classSettings) P() float64 {
+	return *ic.getFloatProperty(topPProperty, &DefaultAnthropicP)
+}
+
+func (ic *classSettings) StopSequences() []string {
+	return *ic.getListOfStringsProperty(stopSequencesProperty, DefaultAnthropicStopSequences)
+}
+
+func contains[T comparable](s []T, e T) bool {
```
Comment (antas-marcin) [inline modules/generative-anthropic/config/class_settings.go:153]: No need to declare this method, you can use `basesettings.ValidateSetting(...)` method, it does the same

File: modules/generative-anthropic/module.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		return errors.Wrap(err, "init q/a")
+	}
+
+	return nil
+}
+
+func (m *GenerativeAnthropicModule) initAdditional(ctx context.Context, timeout time.Duration,
+	logger logrus.FieldLogger,
+) error {
+	apiKey := os.Getenv("ANTHROPIC_API_KEY")
```
Comment (antas-marcin) [inline modules/generative-anthropic/module.go:68]: the naming convention that we use is `ANTHROPIC_APIKEY`

File: adapters/handlers/rest/configure_api.go [hunk 3]
```diff
@@ -926,6 +927,15 @@ func registerModules(appState *state.State) error {
 			Debug("enabled module")
 	}
 
+	if _, ok := enabledModules[modgenerativeanthropic.Name]; ok {
+		appState.Modules.Register(modgenerativeanthropic.New())
+		appState.Logger.
+			WithField("action", "startup").
+			WithField("module", modgenerativeanthropic.Name).
+			Debug("enabled module")
+
```
Comment (antas-marcin) [inline adapters/handlers/rest/configure_api.go:936]: please remove empty line

File: adapters/handlers/rest/configure_api.go [hunk 4]
```diff
@@ -926,6 +927,15 @@ func registerModules(appState *state.State) error {
 			Debug("enabled module")
 	}
 
+	if _, ok := enabledModules[modgenerativeanthropic.Name]; ok {
```
Comment (antas-marcin) [inline adapters/handlers/rest/configure_api.go:930]: also please add this module to `defaultGenerative` list (line 704) as I believe this module can be activated by default bc it's a 3rd party module

File: modules/generative-anthropic/clients/anthropic.go [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+		if keyHeader, ok := value.([]string); ok && len(keyHeader) > 0 && len(keyHeader[0]) > 0 {
+			return keyHeader[0]
+		}
+	}
+	// try getting header from GRPC if not successful
+	if apiKey := modulecomponents.GetValueFromGRPC(ctx, key); len(apiKey) > 0 && len(apiKey[0]) > 0 {
+		return apiKey[0]
+	}
+	return ""
+}
```
Comment (antas-marcin) [inline modules/generative-anthropic/clients/anthropic.go:208]: please use `modulecomponents.GetValueFromContext(...)` method instead

File: modules/generative-anthropic/clients/anthropic.go [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+			Result: &trimmedResponse,
+		}, nil
+	}
+
+	return nil, errors.New("no response from Anthropic")
+}
+
+func (a *anthropic) getAnthropicUrl(ctx context.Context, baseURL string) (string, error) {
+	passedBaseURL := baseURL
+	if headerBaseURL := a.getValueFromContext(ctx, "X-Anthropic-Baseurl"); headerBaseURL != "" {
```
Comment (antas-marcin) [inline modules/generative-anthropic/clients/anthropic.go:159]: please add header value to allowed header [list](https://github.com/weaviate/weaviate/blob/main/usecases/config/config_handler.go#L291)

File: modules/generative-anthropic/clients/anthropic.go [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+	}
+	// try getting header from GRPC if not successful
+	if apiKey := modulecomponents.GetValueFromGRPC(ctx, key); len(apiKey) > 0 && len(apiKey[0]) > 0 {
+		return apiKey[0]
+	}
+	return ""
+}
+
+func (a *anthropic) getApiKey(ctx context.Context) (string, error) {
+	if apiKey := a.getValueFromContext(ctx, "X-Anthropic-Api-Key"); apiKey != "" {
```
Comment (antas-marcin) [inline modules/generative-anthropic/clients/anthropic.go:211]: please add header value to allowed header [list](https://github.com/weaviate/weaviate/blob/main/usecases/config/config_handler.go#L291)

File: modules/generative-anthropic/clients/anthropic.go [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+		return apiKey, nil
+	}
+	if a.apiKey != "" {
+		return a.apiKey, nil
+	}
+	return "", errors.New("no api key found for Anthropic " +
+		"neither in request header: X-Anthropic-Api-Key " +
+		"nor in the environment variable under ANTHROPIC_API_KEY")
```
Comment (antas-marcin) [inline modules/generative-anthropic/clients/anthropic.go:219]: please change to `ANTHROPIC_APIKEY`

File: modules/generative-anthropic/clients/anthropic.go (hunk 6 above)
Comment (cdpierse) [reply modules/generative-anthropic/clients/anthropic.go]: Didn't know this about this one, thanks!

File: adapters/handlers/rest/configure_api.go (hunk 3 above)
Comment (antas-marcin) [reply adapters/handlers/rest/configure_api.go:937]: I think that his empty line is still here

**Pull request merged**